apps/src/lib/node/ledger/shell/prepare_proposal.rs

```diff
@@ -77,6 +64,231 @@ mod prepare_block {
                 ..Default::default()
             }
         }
+
+        /// Builds a batch of vote extension transactions, comprised of Ethereum
+        /// events
+        fn build_vote_extensions_txs(
+            &mut self,
+            local_last_commit: Option<ExtendedCommitInfo>,
+        ) -> Vec<TxRecord> {
+            let protocol_key = self
+                .mode
+                .get_protocol_key()
+                .expect("Validators should always have a protocol key");
+
+            let vote_extension_digest =
+                local_last_commit.and_then(|local_last_commit| {
+                    let votes = local_last_commit.votes;
+                    self.compress_vote_extensions(votes)
+                });
+            let vote_extension_digest = match vote_extension_digest {
+                Some(_) if self.storage.last_height.0 == 0 => {
+                    tracing::error!(
+                        "The genesis block should not contain vote extensions"
+                    );
+                    return vec![];
+                }
+                Some(d) => d,
+                // if no vote extensions were found, we return an empty
+                // `Vec` of protocol
+                // transactions
+                _ => return vec![],
```

Comment:
Apart from genesis, this is a tendermint consensus failure. I think we should put an `unreachable!()` and crash out because something very bad has happened.